### DISCUSSION TITLE
Don't ask Delta question here - ask elsewhere!

### DISCUSSION THREAD

**Discussion by sidhu-patil** [singerdmx/flutter-quill#943] [General]
Hey Brother,
First of all, thank you very much you are a great person.
Brother, I also little bit contributed to your project by creating the **[delta to HTML](https://github.com/secanonm/delta_to_html/)** library 
but I face some problems with block elements. Can you please help me to improve delta to HTML features please make simple delta output like inline elements without "\n"

## Problem 1: delta output return with "\n" hard to predict heading.
![problem1](https://user-images.githubusercontent.com/54258853/189267882-bdf14edb-0592-4954-905f-dc4fda9802c3.jpg)

### output
```JSON
[{"insert":"Hello "},{"insert":"World","attributes":{"bold":true}},{"insert":"\nIt's heading"},{"insert":"\n","attributes":{"header":1}},{"insert":"\n"}] 
```

## Problem 2: when we used inline style on the heading, its returned complex delta output. the heading is unpredictable.
![problem2](https://user-images.githubusercontent.com/54258853/189268608-9a381289-f668-40d0-afed-38746f223f67.jpg)

### output
```JSON
[{"insert":"Hello "},{"insert":"World","attributes":{"bold":true}},{"insert":"\nIt's "},{"insert":"heading","attributes":{"underline":true}},{"insert":"\n","attributes":{"header":1}},{"insert":"\n"}]
```

**Can you please explain your delta output or make it simple**

**Thank you**

**Comment by singerdmx** [comment]
I am still unsure what is needed

**Comment by singerdmx** [comment]
What do you mean by heading?

**Comment by sidhu-patil** [comment]
H1,H2 and H3  Brother

**Comment by singerdmx** [comment]
Look for `attributes":{"header":1}}` which is at the end of a block

**Comment by singerdmx** [comment]
the pattern is 
```
{"insert":"\n"}, XXXXXX {"insert":"\n","attributes":{"header":1}}
```

**Comment by sidhu-patil** [comment]
```JSON
[{"insert":"Hello "},{"insert":"World","attributes":{"bold":true}},{"insert":"\nIt's heading"},{"insert":"\n","attributes":{"header":1}},{"insert":"\n"}] 
```

Brother see this output is complex to predict which is heading [h1,h2,h3] 

I wanna output like 

header/heading[h1,h2,h3] have a in one pair like this


```JSON
[{"insert":"Hello "},{"insert":"World","attributes":{"bold":true}},{"insert":"It's heading","attributes":{"header":1}},{"insert":"\n"}] 
```

**Comment by singerdmx** [comment]
Again

the pattern is 
```
{"insert":"\n"}, XXXXXX {"insert":"\n","attributes":{"header":1}}
```

**Comment by singerdmx** [comment]
Look for `{"insert":"\n","attributes":{"header":1}}`
`header` always comes with `\n`

**Comment by sidhu-patil** [comment]
Brother yeah it's easy when we have only heading[h1,h2,h3] but it is complex when we apply bold, italic,etc styles on headings
See The problem no. 2
[{"insert":"Hello "},{"insert":"World","attributes":{"bold":true}},{"insert":"\nIt's "},{"insert":"heading","attributes":{"underline":true}},{"insert":"\n","attributes":{"header":1}},{"insert":"\n"}]

**Comment by singerdmx** [comment]
the block still ends with `{"insert":"\n","attributes":{"header":1}}`

**Comment by sidhu-patil** [comment]
brother main problem is how we can detect that how much text is heading in map key

**Comment by singerdmx** [comment]
the block starts with `\n` and ends with {"insert":"\n","attributes":{"header":1}}

**Comment by sidhu-patil** [comment]
ok brother great, but here one more problem occurs how we can detect that \n is refer for a block or user press enter for a new line
because block and user enter both are "\n"

**Comment by singerdmx** [comment]
If you see `{"insert":"\n","attributes":{"header":1}}` and from there search upstream for the first `\n`, that defines the block

**Comment by sidhu-patil** [comment]
https://github.com/secanonm/delta_to_html/blob/master/lib/src/encode.dart 
can you please update this code because I'm a beginner
I did not properly get you

**Comment by sidhu-patil** [comment]
brother can you please tell me how to search upstream '\n' in in map key

**Reply by singerdmx** [reply]
Just iterate backwards

**Reply by sidhu-patil** [reply]
can you please provide me a little example

**Comment by singerdmx** [comment]
This has nothing to do with this repo and I am going to close this.
Go ask in any forum related to delta https://stackoverflow.com/questions/39519950/convert-quill-delta-to-html

**Comment by singerdmx** [comment]
This repo uses delta does not mean you should ask here!

**Comment by singerdmx** [comment]
GO check elsewhere like https://github.com/dchenk/go-render-quill
This has nothing to do with this repo.
We only uses delta as underlying data format
